ci: update release process

abmusse opened the pull request:
For more details on the release process, please read docs/RELEASE.md
    
 In previous releases, we packaged the db2ia.node
 and other extra files
    
 Now we restrict the files packaged using the files section
 of the package.json
    
 https://docs.npmjs.com/cli/v8/configuring-npm/package-json#files

kadler reviewed (commented):
Looks good, thought I think some of this stuff could be moved outside of the root. Both the release.sh and README section are really only relevant to us. Perhaps moving the docs to docs/RELEASE.md and putting release.sh under .github?

abmusse commented:
That sounds good to me as well!

abmusse commented:
polite ping  @kadler

> Looks good, thought I think some of this stuff could be moved outside of the root. Both the release.sh and README section are really only relevant to us. Perhaps moving the docs to docs/RELEASE.md and putting release.sh under .github?

Can you please re-review after I moved files around as this PR is now going stale 😄

The steps for the release process are defined in 
https://github.com/IBM/nodejs-idb-connector/blob/44065d373c458ae697294f9dee8ba9956bbc7ee9/docs/RELEASE.md
Let me know if this needs some changes.

abmusse committed:
ci: update release process

For more details on the release process, please read docs/RELEASE.md

In previous releases, we packaged the db2ia.node
and other extra files

Now we restrict the files packaged using the files section
of the package.json

https://docs.npmjs.com/cli/v8/configuring-npm/package-json#files
